src/app/hooks/useClickAway.ts

```diff
@@ -0,0 +1,37 @@
+"use client";
+
+import { useEffect, useRef } from "react";
+
+const events = ["mousedown", "touchstart"];
+
+const useClickAway = <T extends HTMLElement>(handler: (e: Event) => void) => {
+  const ref = useRef<T>(null);
+
+  const savedHandler = useRef(handler);
+  useEffect(() => {
+    savedHandler.current = handler;
+  }, [handler]);
+
+  useEffect(() => {
+    const element = ref.current;
+    if (!element) return;
+
+    const handleEvent = (e: Event) => {
+      const targetNode = e.target as Node;
+      !element.contains(targetNode) && savedHandler.current(e);
+    };
```

Comment:
타입 단언 사용보다는 타입 가드를 사용해서 사이드 이펙트를 방지하는건 어떠실까요?
```suggestion
 const handleEvent = (e: Event) => {
   if (e.target instanceof Node) {
      const targetNode = e.target;
      !element.contains(targetNode) && savedHandler.current(e);
       }
    };
```

Reply:
아하 좋은지적 감사합니다 바로 반영해보겠습니다!